app/controllers/homes_controller.rb

```diff
@@ -3,7 +3,8 @@ def show
     if signed_out?
       render "pages/landing"
     else
-      @recommended_items = RecommendedItem.all
+      following_and_self = current_user.following + [current_user]
+      @recommended_items = RecommendedItem.where(user: following_and_self)
```

Comment:
Usually I _love_ utility classes, but since this is so close to being `user.following` (an AR relation) anyway, I'll put it on User. But I have my eye on you, _User_!